src/ApiConnectors/ArticleApiConnector.php

```diff
@@ -91,4 +88,57 @@ public function sendAll(array $articles): MappedResponseCollection
             return ArticleMapper::map($response);
         });
     }
-}
+
+	/**
+     * List all articles.
+     *
+     * @param string $pattern  The search pattern. May contain wildcards * and ?
+     * @param int    $field    The search field determines which field or fields will be searched. The available fields
+     *                         depends on the finder type. Passing a value outside the specified values will cause an
+     *                         error.
+     * @param int    $firstRow First row to return, useful for paging
+     * @param int    $maxRows  Maximum number of rows to return, useful for paging
+     * @param array  $options  The Finder options. Passing an unsupported name or value causes an error. It's possible
+     *                         to add multiple options. An option name may be used once, specifying an option multiple
+     *                         times will cause an error.
+     *
+     * @return Article[] The articles found.
+     */
+    public function listAll(
+        string $pattern = '*',
+        int $field = 0,
+        int $firstRow = 1,
+        int $maxRows = 100,
+        array $options = []
+    ): array {
+        $optionsArrayOfString = array('ArrayOfString' => array());
+
+        foreach ($options as $key => $value) {
+            $optionsArrayOfString['ArrayOfString'][] = array($key, $value);
+        }
+
+        $response = $this->getFinderService()->searchFinder(FinderService::TYPE_ITEMS, $pattern, $field, $firstRow, $maxRows, $optionsArrayOfString);
+
+        if ($response->data->TotalRows == 0) {
+            return [];
+        }
+
+        $articles = [];
+
+        foreach ($response->data->Items->ArrayOfString as $articleArray) {
```

Comment:
I see this bit of code again and again in the PRs. Should this be moved to a private method or a helper trait?

Reply:
Fixed in #142